### ISSUE TITLE
Security API does not throw proper error messages

### ISSUE THREAD

**Issue by dutchmo** [thisbejim/Pyrebase#46]
The API does not seem to report error codes on security operations. I believe closed issue #44 may be an example of this.

There are certain validations that occur on the firebase end, for example, `createUserWithEmailAndPassword` can fail for weak password (less than 6 characters) or invalid email (missing an '@' for example).

For these cases, the API seems to just throw a generic 400 error, like:

`requests.exceptions.HTTPError: 400 Client Error: Bad Request for url: https://www.googleapis.com/identitytoolkit/v3/relyingparty/verifyPassword?key=AIzaS...`

It is not clear whether these errors are even getting sent back from firebase, in that case it might be helpful to do a couple validations on the client side, or simply document the issue.

**Comment by thisbejim** [comment]
Thanks @dutchmo, I totally agree, I remember adding more detailed error messages for the get method, for example:
```
try:
    request_object.raise_for_status()
except HTTPError as e:
    # raise detailed error message
    raise HTTPError(e, request_object.text)
```
Will add these for other methods too.

**Comment by thisbejim** [comment]
Fixed with 72ba14519937220be4297a17c64647524222facb
